src/relentless/model/potential/potential.py

```diff
@@ -490,3 +490,319 @@ def _derivative(self, param, r, **params):
 
         """
         pass
+
+
+class BondedSpline(BondedPotential):
+    """Base class for bonded spline potentials.
+
+    The bonded spline potential is defined by interpolation through a set of
+    knot points. The interpolation scheme uses Akima splines.
+
+    This class should not be instantiated directly. Instead, use the appropriate
+    spline type, i.e., :class:`~relentless.model.potential.BondSpline` or
+    :class:`~relentless.model.potential.AngleSpline`.
+
+    Parameters
+    ----------
+    types : tuple[str]
+        Types.
+    num_knots : int
+        Number of knots.
+    mode : str
+        Mode for storing the values of the knots in
+        :class:`~relentless.variable.Variable` that can be optimized. If
+        ``mode='value'``, the knot amplitudes are manipulated directly.
+        If ``mode='diff'``, the amplitude of the *last* knot is fixed, and
+        differences between neighboring knots are manipulated for all other
+        knots. Defaults to ``'diff'``.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    """
+
+    valid_modes = ("value", "diff")
+    _space_coord_name = "x"
```

Comment:
I noticed that the parent `BondedPotential` class uses `r` as its space variable for `derivative`. I would go replace that by `x` to make this consistent in our code base.